diff --git a/common/common.cpp b/common/common.cpp
index d8319cd9a..e9b4b36fe 100644
--- a/common/common.cpp
+++ b/common/common.cpp
@@ -1747,6 +1747,7 @@ struct llama_context_params common_context_params_to_llama(const common_params &
     cparams.op_offload        = !params.no_op_offload;
     cparams.swa_full          = params.swa_full;
     cparams.kv_unified        = params.kv_unified;
+    cparams.gpu_pill          = params.gpu_pill;
 
     cparams.type_k = params.cache_type_k;
     cparams.type_v = params.cache_type_v;
