diff --git a/src/llama-cparams.h b/src/llama-cparams.h
index b592de18c..e67d793f6 100644
--- a/src/llama-cparams.h
+++ b/src/llama-cparams.h
@@ -52,6 +52,7 @@ struct llama_cparams {
     bool warmup;             // TODO: remove [TAG_LLAMA_GRAPH_NO_WARMUP]
     bool op_offload;
     bool kv_unified;
+    bool gpu_pill;
     bool pipeline_parallel;
 
     std::vector<bool> embeddings_layer_inp; // [n_layer()] extract input embeddings for layer
