diff --git a/src/llama-cparams.h b/src/llama-cparams.h
index f4eb181..6e36a64 100644
--- a/src/llama-cparams.h
+++ b/src/llama-cparams.h
@@ -52,6 +52,7 @@ struct llama_cparams {
     bool warmup;             // TODO: remove [TAG_LLAMA_GRAPH_NO_WARMUP]
     bool op_offload;
     bool kv_unified;
+    bool gpu_pill;
     bool pipeline_parallel;
     bool training;           // set by llama_opt_init()
 
