diff --git a/src/llama-kv-cache-msa.h b/src/llama-kv-cache-msa.h
index f09b6d32b..9c312a89a 100644
--- a/src/llama-kv-cache-msa.h
+++ b/src/llama-kv-cache-msa.h
@@ -20,6 +20,7 @@ public:
                          bool   v_trans,
                          bool   offload,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_pad,
