--- a/src/llama-memory-hybrid-idx.h	2026-09-22 00:26:37.287114542 -0700
+++ b/src/llama-memory-hybrid-idx.h	2026-09-22 00:26:37.287131664 -0700
@@ -33,6 +33,7 @@
                  uint32_t   n_rs_seq,
                      bool   offload,
                      bool   unified,
+                     bool   gpu_pill,
                             /* layer filters */
     const layer_filter_cb & filter_attn,
     const layer_filter_cb & filter_recr,
