diff --git a/src/llama-kv-cache-dsa.h b/src/llama-kv-cache-dsa.h
index e74fc4d91..5075937ba 100644
--- a/src/llama-kv-cache-dsa.h
+++ b/src/llama-kv-cache-dsa.h
@@ -21,6 +21,7 @@ public:
                          bool   v_trans,
                          bool   offload,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_pad,
