--- a/src/llama-memory-hybrid.h	2026-09-22 00:26:37.280784101 -0700
+++ b/src/llama-memory-hybrid.h	2026-09-22 00:26:37.280801003 -0700
@@ -37,6 +37,7 @@
                  uint32_t   n_rs_seq,
                      bool   offload,
                      bool   unified,
+                     bool   gpu_pill,
                             /* layer filters */
     const layer_filter_cb & filter_attn = nullptr,
     const layer_filter_cb & filter_recr = nullptr);
