diff --git a/src/llama-kv-cache-iswa.h b/src/llama-kv-cache-iswa.h
index 7dab6eaa8..84da10b27 100644
--- a/src/llama-kv-cache-iswa.h
+++ b/src/llama-kv-cache-iswa.h
@@ -21,6 +21,7 @@ public:
                          bool   offload,
                          bool   swa_full,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_ubatch,
@@ -39,6 +40,7 @@ public:
                          bool   offload,
                          bool   swa_full,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_ubatch,
