diff --git a/src/llama-kv-cache.h b/src/llama-kv-cache.h
index c4d8699de..436be0f0f 100644
--- a/src/llama-kv-cache.h
+++ b/src/llama-kv-cache.h
@@ -104,6 +104,7 @@ public:
                          bool   v_trans,
                          bool   offload,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_pad,
