diff --git a/src/llama-model.cpp b/src/llama-model.cpp
index d87481381..973233ded 100644
--- a/src/llama-model.cpp
+++ b/src/llama-model.cpp
@@ -2316,6 +2316,10 @@ llama_model_params llama_model_default_params() {
         /*.use_extra_bufts             =*/ true,
         /*.no_host                     =*/ false,
         /*.no_alloc                    =*/ false,
+#ifdef GUANACO_ENABLED
+        /*.use_guanaco_streaming       =*/ false,
+        /*.guanaco_max_experts         =*/ -1,
+#endif
     };
 
     return result;
