diff --git a/src/llama-memory-hybrid-idx.h b/src/llama-memory-hybrid-idx.h
index 0ac4265..952f0e7 100644
--- a/src/llama-memory-hybrid-idx.h
+++ b/src/llama-memory-hybrid-idx.h
@@ -35,6 +35,7 @@ public:
                  uint32_t   n_rs_seq,
                      bool   offload,
                      bool   unified,
+                     bool   gpu_pill,
                             /* layer filters */
     const layer_filter_cb & filter_attn,
     const layer_filter_cb & filter_recr,
