diff --git a/src/llama-kv-cache-dsv4.h b/src/llama-kv-cache-dsv4.h
index ce39867c0..92e7394c5 100644
--- a/src/llama-kv-cache-dsv4.h
+++ b/src/llama-kv-cache-dsv4.h
@@ -95,6 +95,7 @@ public:
                          bool   offload,
                          bool   swa_full,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_ubatch,
