--- a/src/llama-kv-cache-msa.h	2026-09-22 00:26:37.272494213 -0700
+++ b/src/llama-kv-cache-msa.h	2026-09-22 00:26:37.272508029 -0700
@@ -20,6 +20,7 @@
                          bool   v_trans,
                          bool   offload,
                          bool   unified,
+                         bool   gpu_pill,
                      uint32_t   kv_size,
                      uint32_t   n_seq_max,
                      uint32_t   n_pad,
